libs/api/ebpf_api.cpp

```diff
@@ -1626,17 +1634,31 @@ _ebpf_program_attach(
     }
 
     ebpf_result_t result = _link_ebpf_program(
-        program_handle, attach_type, link_handle, (uint8_t*)attach_parameters, attach_parameters_size);
+        program_handle, attach_type, &local_link_handle, (uint8_t*)attach_parameters, attach_parameters_size);
     if (result != EBPF_SUCCESS) {
         EBPF_RETURN_RESULT(result);
     }
 
-    if (link_fd != nullptr) {
-        *link_fd = _create_file_descriptor_for_handle(*link_handle);
-        if (*link_fd == ebpf_fd_invalid) {
-            Platform::CloseHandle(*link_handle);
-            *link_handle = ebpf_handle_invalid;
-            EBPF_RETURN_RESULT(EBPF_NO_MEMORY);
+    if (link_handle == nullptr && link_fd == nullptr) {
+        result = _ebpf_link_mark_as_legacy_mode(local_link_handle);
+        Platform::CloseHandle(local_link_handle);
+        local_link_handle = ebpf_handle_invalid;
+        if (result != EBPF_SUCCESS) {
+            EBPF_RETURN_RESULT(result);
+        }
+
```

Comment:
```suggestion
```
If you update this PR for some other reason.  No need to update just for this though.